src/compiler/transformers/module/esnextAnd2015.ts

```diff
@@ -238,9 +242,27 @@ export function transformECMAScriptModule(context: TransformationContext): (x: S
         return statements;
     }
 
-    function visitExportAssignment(node: ExportAssignment): VisitResult<ExportAssignment | undefined> {
-        // Elide `export=` as it is not legal with --module ES6
-        return node.isExportEquals ? undefined : node;
+    function visitExportAssignment(node: ExportAssignment): VisitResult<ExportAssignment | ExpressionStatement | undefined> {
+        if (node.isExportEquals) {
+            if (getEmitModuleKind(compilerOptions) === ModuleKind.Preserve) {

```

Comment:
At some level I wish this were written like this to reduce nesting:

```ts
if (!node.isExportEquals) {
	return node;
}
if (getEmitModuleKind(compilerOptions) !== ModuleKind.Preserve) {
	return undefined;
}
// big long text
```

But that's a silly nit and we aren't consistent about early returns anyway.